pkg/gofr/datasource/file/ftp/interface.go

```diff
@@ -32,6 +34,9 @@ type ServerConn interface {
 	RemoveDir(path string) error
 	Quit() error
 	FileSize(name string) (int64, error)
+	CurrentDir() (string, error)
+	ChangeDir(path string) error
+	List(path string) ([]*ftp.Entry, error)
```

Comment:
I would defined a type alias to do not expose the ftp lib outside the project.

Reply:
Are you referring to defining a local interface similar to how I did for FileInfo ?

Reply:
It could be ether:
- an interface (if it's possible)
- an alias

An alias would be this:

`type Entry *ftp.Entry`

This way in the List method you loop on the ftp.Entry to create a list of Entry

This way the caller only knows about the gofr ftp package, and not about the external ftp library

Reply:
I have a doubt regarding this implementation. Revisiting the code, I realised that..... what the user is handed over through Stat or ReadDir is FileInfo Interface that allows the user to call methods defined on them. 

They will never be  exposed to ftp.Entry, because I add all the information that ftp.Entry returns in the file struct itself inside the directory functions. This file struct implements the FileInfo interface. So is there really a need to define a type like this for ftp.Entry to mask it ? 

What is the benefit of this, that I might be missing ?

Reply:
The servercon interface is exported.

But I conceded the servercon might not be used outside gofr implementation.

I'm OK with doing without the type alias

It's not a real problem also.

So yes, you can rollback  and clean as you did

Reply:
I saw you changed the visibility of ServerConn

The interface was already there, so it was previously exported.

It might be OK, and better.

I would like to get a feedback from 

@aryanmehrotra @vipul-rawat @srijan-27 @Umang01-hash

Reply:
It should be fine if it is unexported, because it was not used before as well by user. As user would have access to the File interface and File will be used testing. So, I guess it should be fine being unexported.

P.S: We should be vigilant more on what has to be exported going forward, so we don't end up changing later - Starting with keeping unexported is better @ccoVeille @coolwednesday

Reply:
Thanks for your feedback.

I now tend to use internal packages when starting a new project, it avoids the classic debate about "how can keep this struct unexported but use it in another package of the same repository" sometimes it's impossible to solve the issue because on cyclic import it may lead, so using internal help. But it's easier to start with it, than adding it to an existing project with large code base